petastorm/workers_pool/process_pool.py

```diff
@@ -42,9 +41,36 @@
 # recheck if no more items are expected to be ventilated
 _VERIFY_END_OF_VENTILATION_PERIOD = 0.1
 
+_WORKER_STARTED_INDICATOR = 'worker started indicator'
+
 logger = logging.getLogger(__name__)
 
 
+#
+# ----------------                                    ------------------
+# |              |  --- _ventilator_send  (push) -->  |                |
+# | main process |  --- _control_sender   (pub)  -->  | worker process |
+# |              |  <-- _results_receiver (pull)  --  |                |
+# ----------------                                    ------------------
+#
+# 1. When ProcessPool start is called, it creates _ventilator_send, _control_sender and _result_receiver
+#    sockets.
+# 2. After initialization is done, worker process sends _WORKER_STARTED_INDICATOR
```

Comment:
If the _WORKER_STARTED_INDICATOR is sent by the worker before the results_receiver socket is setup is it possible to lose the indicator causing blocking/timeouts. Im not sure if thats a possibility but its something I vaguely remember seeing. I think its something we can keep in mind if we start seeing failures for that.